### PR TITLE
ftests: add functional tests

### PR DESCRIPTION
**Pull request by lucab**

### CODE REVIEW

File: ftests/generator_test.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+package ftests
+
+import (
+	"os"
+	"os/exec"
+	"testing"
+)
+
+func TestGeneratorEmpty(t *testing.T) {
+	if !IsInContainer() {
```
Comment (squeed) [inline ftests/generator_test.go]: It's like forking, but better!

File: ftests/smoke_test.go [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+			imageName: EmptyImage,
+		}
+		RunTestInContainer(t, cfg)
+		return
+	}
+
+	cmd := exec.Command("torcx", "clearly-bogus")
+	err := cmd.Run()
+	if err == nil {
+		t.Error("expected error, got nil")
```
Comment (squeed) [inline ftests/smoke_test.go]: Probably want to check that the error code is not 253, which is rkt's exit code.

Comment (lucab) [inline ftests/smoke_test.go]: Inception mode: everything after the initial `if` block is run directly by testrunner **inside** a rkt container, so this line will never seen any rkt error as is it only executed inside it. On the other hand, `RunTestInContainer` will propagate both rkt error and tests failure back to the original runner.

Comment (squeed) [inline ftests/smoke_test.go]: ahhhhhh, got that mixed up. Thanks!

**Pull request merged**